cryptographic_flows.md

```diff
@@ -8,28 +8,25 @@ An asset owner that has not previously interacted with the key server MUST regis
     1. Generates `user_id`, a globally unique identifier (GUID). 
         - [TODO #52](https://github.com/boltlabs-inc/key-mgmt-spec/issues/52): Specify how user IDs are created.
     1. [Opens a registration session](systems-architecture.md#opening-a-registration-session) with the key server.
-    1. Derives `master_key`, a symmetric key of length `len` bytes for [`Enc`](#external-dependencies), from  `export_key`, where the length `len` should be the default for `Enc`, as follows:
-        1. Set `context = "Lock Keeper master key "`.
-            - [TODO #27](https://github.com/boltlabs-inc/key-mgmt-spec/issues/27) Insert AEAD parameter details.
-        1. Compute `secret = HKDF(export_key, "opaque-derived master key" | context)`.
-        1. Define `master_key = (secret, context)`.
+    1. Derives `master_key`, a symmetric key of length `len` bytes for [`Enc`](#external-dependencies), from  `export_key`, as follows:
+        1. Length `len` should be the default for `Enc`.
+        1. Set `master_key = HKDF(export_key, "OPAQUE-derived Lock Keeper master key")`.
     1. Runs the [generate protocol](cryptographic_flows.md#generate-a-secret) to get a symmetric key `storage_key` for [`Enc`](#external-dependencies) of length 32 bytes.
         - The `storage key` MUST NOT be saved, stored, or used in any context outside this protocol. It must not be passed to the calling application.
-    1. Computes a ciphertext `encrypted_storage_key = Enc(opaque_encryption_key, storage_key, user_id||"storage key")`.
+    1. Computes a ciphertext `encrypted_storage_key = Enc(master_key, storage_key, user_id||"storage key"|context)`.
```

Comment:
Open question: Is there additional context here beyond "storage key"? We temporarily had "context" set in a previous step, but that has been removed now. The only thing I can think of is details of the encryption scheme or maybe a note that it was encrypted with the master key, but neither of those seem particularly useful to me since you have to know them to decrypt the thing anyway.

Reply:
I had simplified this for now, based on your previous comments and after thinking about how we plan to use this key. 

In general, though, it's much safer to view a key as both the key material and the context (which primitive, which parameters) and to be careful to enforce that the context is correct. This helps prevent downstream misuse like where you use the key in a different primitive or with different parameters, for example as part of an attack. Does this make sense? @marsella